server/form-pages/assess/reviewApplication/reviewApplicationAndDocumentsPage.test.ts

```diff
@@ -14,7 +22,42 @@ describe('Review', () => {
     })
   })
 
-  itShouldHaveNextValue(new Review({}), '')
+  describe('initialize', () => {
+    beforeEach(() => {
+      assessment.application = overwriteApplicationDocuments(assessment.application, [])
+    })
+
+    it('should call the API to fetch documents for the application', async () => {
+      const getDocumentsMock = jest.fn().mockResolvedValue(documentFactory.buildList(3))
+      const applicationService = createMock<ApplicationService>({ getDocuments: getDocumentsMock })
+
+      await Review.initialize({}, assessment, token, { applicationService })
+
+      expect(applicationService.getDocuments).toHaveBeenCalledWith(token, assessment.application)
+    })
+
+    it('should add the full Document objects given the document filenames supplied on the application', async () => {
+      const fullSelectedDocumentObjects = [
+        documentFactory.build({ fileName: 'documentA' }),
+        documentFactory.build({ fileName: 'documentB' }),
+      ]
+      assessment.application = overwriteApplicationDocuments(assessment.application, [
+        { fileName: 'documentA' } as Document,
+        { fileName: 'documentB' } as Document,
+      ])
+
+      const getDocumentsMock = jest.fn().mockResolvedValue(fullSelectedDocumentObjects)
+
+      const applicationService = createMock<ApplicationService>({ getDocuments: getDocumentsMock })
+
+      const page = await Review.initialize({}, assessment, token, { applicationService })
+
+      // eslint-disable-next-line
+      expect(documentsFromApplication((page as any).document.application)).toEqual(fullSelectedDocumentObjects)
```

Comment:
I'd usually expect this to be the other way round unless there's a specific reason? Also, would be good to have the results of `documentsFromApplication` as a variable, so:

```suggestion
      const expectedDocuments = documentsFromApplication((page as any).document.application)
      expect(fullSelectedDocumentObjects).toEqual(expectedDocuments)
```

Reply:
I agree with having the result of `documentsFromApplication` as a variable but what you have called `expectedDocuments` is the thing we want to assert is like the `fullSelectedDocumentObjects` so I think it should be: 
``` const actualDocuments = documentsFromApplication((page as any).document.application)
      expect(actualDocuments).toEqual(fullSelectedDocumentObjects)
```
If that makes sense to you? I'm wondering if there is something I could do to make it a bit clearer. Maybe using the `documentsFromApplication` helper confuses things here?

Reply:
How about this?:

```
    it('should add the full Document objects given the document filenames supplied on the application', async () => {
      const allDocuments = [
        documentFactory.build({ fileName: 'documentA' }),
        documentFactory.build({ fileName: 'documentB' }),
        documentFactory.build({ fileName: 'documentC' }),
      ]
      assessment.application = overwriteApplicationDocuments(assessment.application, [
        { fileName: allDocuments[0].fileName } as Document,
        { fileName: allDocuments[1].fileName } as Document,
      ])

      const getDocumentsMock = jest.fn().mockResolvedValue(allDocuments)

      const applicationService = createMock<ApplicationService>({ getDocuments: getDocumentsMock })

      const page = await Review.initialize({}, assessment, token, { applicationService })

      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const actualDocuments = documentsFromApplication((page as any).document.application)

      expect(actualDocuments).toEqual([allDocuments[0], allDocuments[1]])
    })
```

Reply:
This way you're clearer that what you want to get back is a subset of all the documents that you're mocking.

Reply:
Yep that's better